x/authz/client/cli/tx.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+	authclient "github.com/cosmos/cosmos-sdk/x/auth/client"
+	"github.com/cosmos/cosmos-sdk/x/authz/types"
+)
+
+// GetTxCmd returns the transaction commands for this module
+func GetTxCmd() *cobra.Command {
+	AuthorizationTxCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "Authorization transactions subcommands",
+		Long:                       "Authorize and revoke access to execute transactions on behalf of your address",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	AuthorizationTxCmd.AddCommand(
+		NewCmdGrantAuthorization(),
+		NewCmdRevokeAuthorization(),
+		NewCmdExecAuthorization(),
+	)
+
+	return AuthorizationTxCmd
+}
+
+func NewCmdGrantAuthorization() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "grant [grantee_address] [msg-type] [limit] --from=[granter]",
+		Short: "Grant authorization to an address",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Grant authorization to an address to execute a transaction on your behalf:
+
+Examples:
+ $ %s tx %s grant cosmos1skjw.. %s 1000stake --from=cosmos1skl..
+ $ %s tx %s grant cosmos1skjw.. /cosmos.gov.v1beta1.Msg/Vote --from=cosmos1sk..
+	`, version.AppName, types.ModuleName, types.SendAuthorization{}.MethodName(), version.AppName, types.ModuleName),
+		),
+		Args: cobra.RangeArgs(2, 3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			grantee, err := sdk.AccAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			msgType := args[1]
+
+			var authorization types.Authorization
+			if msgType == (types.SendAuthorization{}.MethodName()) {
+				limit, err := sdk.ParseCoinsNormalized(args[2])
+				if err != nil {
+					return err
+				}
+				authorization = &types.SendAuthorization{
+					SpendLimit: limit,
+				}
+			} else {
+				authorization = types.NewGenericAuthorization(msgType)
+			}
+
+			exp, err := cmd.Flags().GetInt64(FlagExpiration)
+			if err != nil {
+				return err
+			}
+			period := time.Duration(exp) * time.Second
+
+			msg, err := types.NewMsgGrantAuthorization(clientCtx.GetFromAddress(), grantee, authorization, time.Now().Add(period))
+			if err != nil {
+				return err
+			}
+
+			svcMsgClientConn := &serviceMsgClientConn{}
+			authzMsgClient := types.NewMsgClient(svcMsgClientConn)
+			_, err = authzMsgClient.GrantAuthorization(context.Background(), msg)
+			if err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), svcMsgClientConn.msgs...)
+
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	cmd.Flags().Int64(FlagExpiration, int64(3600*24*365), "The second unit of time duration which the authorization is active for the user; Default is a year")
```

Comment:
I'm not sure using seconds from user is the best way, it may lead to different behaviour based on how long transaction stays in the mempool(minimal as of now).

Can we accept staandard time and calculate seconds to the same

Reply:
Ah, it should be unix time I believe. @aleem1314 ?

Reply:
would be nice to use a more human readable format like `2020-11-18T09:03:47` and parse to unix timestamp internally